apps/client/src/app/pages/inventory-optimizer/optimizations/inventory-optimizer.ts

```diff
@@ -38,6 +38,12 @@ export abstract class InventoryOptimizer {
     return this._getOptimization(item, inventory, data);
   }
 
+  /// Compare two InventoryItem objects to determine whether they consume the same slot in the same
+  /// container.
+  protected static inSameSlot(source: InventoryItem, target: InventoryItem): boolean {
```

Comment:
Static methods are really not recommended in a Typescript context, especially for frontend with dependency injection. It makes it pretty much impossible to mock properly in a testing context, as typescript creates separate objects for static and dynamic contexts.

Reply:
It's a function that doesn't require object state, nor is it something that should be polymorphic. In C++ something like this would be a global or namespace-level free function. What's the philosophy and/or idiom in Typescript? For something like this I don't see how dependency injection is relevant.

Reply:
The issue with static methods is that they can't be mocked at all in a TS context, that's why people tend to avoid static methods in TS. But this one can be fine as it's purely a helper.

Reply:
However, if we keep this as static, it has to be the first field of the class, there's a lint rule for that.

Reply:
I'm not exactly fixated on it being static. That's a carry-over from C++ I learned related to OOP. If a method doesn't need state in a class, make it static. It's a way to promote separation of concerns / encapsulation. If that doesn't apply in Typescript I'm open to an alternative, just make a suggestion. If you have nothing I'll go ahead and move it to the top and we'll move on. Thanks!